File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
      
    	int[] divisors = new int[number];
		
		for(int i=0; i<number; i++) {
			divisors[i] = divisorCnt(i+1);
		}
		
		int sum = 0;

		for(int i=0; i<number; i++) {
			if(divisors[i]>limit) {
				divisors[i]=power;
			}
			sum += divisors[i];
		}
		
        int answer = sum;
		return answer;
	}

	public static int divisorCnt(int n) {
		int count = 0;
		for(int i=1; i<=Math.sqrt(n); i++) {
			if(i==Math.sqrt(n)) count++;
			else if(n%i==0) count+=2; 
		}
		return count;
	}
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    weapons = [0] * number
    for i in range(number):
        knight = i+1
        edge = knight**0.5
        weapon = 0
        for j in range(1, int((edge+1)//1)):
            if j < edge and knight % j == 0:
                weapon += 2
            if j == edge and edge % 1 == 0:
                weapon += 1
        if weapon > limit:
            weapons[i] = power
        else:
            weapons[i] = weapon
    return sum(weapons)